### ISSUE TITLE
what's the differences between the three stage of training

### ISSUE THREAD

**Issue by cain4318** [scaelles/OSVOS-TensorFlow#20]
I have noticed there are three parameters of the training iterations(15000,30000,50000) which divide the training process into three stage:
stage 1 use strong side output supervision
stage 2 use weak.....
stage 3 use no......
except the loss, what's the differences between the three?

**Comment by scaelles** [comment]
Hello,

The only difference in the three stages as you said is the loss. We are basically reducing the influence of the side outputs loss progressively. In stage 1, we multiple them by 1. In stage 2, we multiply them by 0.5 and in stage 3, we multiply them by 0. You can find this in the [code](https://github.com/scaelles/OSVOS-TensorFlow/blob/6ef61d1f523296b95974a6dafac01e94bf2fde7d/osvos.py#L457).

Regarding your first question about the code for OSVOS-S, we plan to release it at some point but it's not a high priority task in our to do list as it is quite similar to OSVOS code.

Let us know if you have any further question.

**Comment by cain4318** [comment]
Thanks a lot for your response!
